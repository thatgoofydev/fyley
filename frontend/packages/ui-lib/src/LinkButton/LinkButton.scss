@import "src/base-styling/vars";

.link-button {
  display: flex;
  width: fit-content;
  text-decoration: none;

  &:hover {
    cursor: pointer;

    span {
      color: darken($btn-text-color, 10%);
      text-decoration: underline;
    }
  }

  .icon {
    margin: 0 4px 0 0;
    transform: translateY(1px);
  }

  span {
    display: inline-block;
    color: $btn-text-color;
  }
}

a.link-button {
  display: inline-block;
}