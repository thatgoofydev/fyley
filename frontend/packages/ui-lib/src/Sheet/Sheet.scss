@import "src/base-styling/vars";

.sheet {
  background-color: $sheet-bg;
  border: $sheet-border;
  border-radius: $sheet-border-radius;
}

.sheet__header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: $sheet-padding;
}

.sheet__content {
  margin: $sheet-padding;
}