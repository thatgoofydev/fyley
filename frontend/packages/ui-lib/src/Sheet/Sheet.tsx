import React, { FC, FunctionComponent, ReactNode } from "react";
import "./Sheet.module.scss";

/* Sheet */

export const Sheet: FunctionComponent = ({ children }) => {
  return <div className="sheet">{children}</div>;
};

/* Header */

interface ISheetHeaderProps {
  title: string;
  buttons?: ReactNode;
}

export const SheetHeader: FunctionComponent<ISheetHeaderProps> = ({ title, buttons }) => {
  return (
    <>
      <div className="sheet__header">
        <h1>{title}</h1>
        {buttons && <div>{buttons}</div>}
      </div>
    </>
  );
};

/* Content */

export const SheetContent: FC = ({ children }) => {
  return <div className="sheet__content">{children}</div>;
};
