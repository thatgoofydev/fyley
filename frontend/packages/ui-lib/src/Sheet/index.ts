export { Sheet, SheetHeader, SheetContent } from "./Sheet";
