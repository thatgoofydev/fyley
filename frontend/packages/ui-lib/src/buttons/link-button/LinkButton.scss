@import "src/base/colors";

.link-button {
  display: flex;
  width: fit-content;
  text-decoration: none;

  &:hover {
    cursor: pointer;

    span {
      color: darken($btn-text-color, 10%);
      text-decoration: underline;
    }
  }

  .icon {
    margin: 1px 4px 0 0;
  }

  span {
    display: inline-block;
    color: $btn-text-color;
  }
}

a.link-button {
  display: inline-block;
}