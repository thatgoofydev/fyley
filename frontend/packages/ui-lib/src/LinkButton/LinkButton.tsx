import React, { FunctionComponent } from "react";
import classNames from "classnames";
import { Icon, IconType } from "../Icon";
import "./LinkButton.scss";

export interface IProps {
  text: string;
  icon?: IconType;
  href?: string;
}

export const LinkButton: FunctionComponent<IProps> = ({
  text,
  icon = undefined,
  href = undefined
}) => {
  const classes = classNames("link-button");

  const renderContent = () => {
    return (
      <>
        {icon && <Icon type={icon} size={14} />}
        <span>{text}</span>
      </>
    );
  };

  const renderAnchor = () => {
    return (
      <a className={classes} href={href}>
        {renderContent()}
      </a>
    );
  };

  const renderButton = () => {
    return (
      <div className={classes} role="button">
        {renderContent()}
      </div>
    );
  };

  return <>{href ? renderAnchor() : renderButton()}</>;
};
