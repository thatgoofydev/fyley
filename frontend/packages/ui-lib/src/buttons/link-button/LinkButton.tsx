import React, { FunctionComponent } from "react";
import classNames from "classnames";
import { Icon, IconType } from "../../icons";
import "./LinkButton.scss";

export interface LinkButtonProps {
  text: string;
  icon?: IconType;
  href?: string;
}

export const LinkButton: FunctionComponent<LinkButtonProps> = ({
  text,
  href = undefined,
  icon = undefined
}) => {
  const classes = classNames("link-button");

  const renderContent = () => {
    return (
      <>
        {icon && <Icon type={icon} size={14} />}
        <span>{text}</span>
      </>
    );
  };

  const renderAnchor = () => {
    return (
      <a className={classes} href={href}>
        {renderContent()}
      </a>
    );
  };

  const renderButton = () => {
    return (
      <div className={classes} role="button">
        {renderContent()}
      </div>
    );
  };

  return <>{href ? renderAnchor() : renderButton()}</>;
};
