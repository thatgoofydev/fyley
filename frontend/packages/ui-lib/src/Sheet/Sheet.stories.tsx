import React from "react";
import { Meta, Story } from "@storybook/react";
import { Sheet, SheetContent, SheetHeader } from "./Sheet";
import { LinkButton } from "../LinkButton";

export default {
  title: "General/Sheet",
  component: Sheet,
  subcomponents: {
    SheetHeader,
    SheetContent
  }
} as Meta;

const FullSheet: Story = _ => {
  return (
    <Sheet>
      <SheetHeader
        title="Sheet Title"
        buttons={<LinkButton text="Add something" icon="add" />}
      />
      <SheetContent>
        <p>Some content inside of the sheet...</p>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad aperiam
          assumenda commodi, consectetur, cupiditate et illo, laborum minus
          nihil officia porro provident quidem quisquam ratione recusandae sit
          ut? Perspiciatis, voluptas.
        </p>
        <p>
          A atque consequatur consequuntur cupiditate deleniti dolorem earum
          omnis, quasi quis ratione sed, tempora veritatis voluptas! A ad animi
          beatae dolor dolore doloremque dolores facilis ipsam, nesciunt odio
          quasi quisquam.
        </p>
      </SheetContent>
    </Sheet>
  );
};
FullSheet.storyName = "Sheet";

export { FullSheet };
