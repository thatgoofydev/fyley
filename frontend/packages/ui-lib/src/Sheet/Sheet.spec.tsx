import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Sheet, SheetContent, SheetHeader } from "./Sheet";
import { LinkButton } from "../LinkButton";

afterEach(cleanup);

describe("Loader", function () {
  test("should render", () => {
    const rendered = render(
      <Sheet>
        <SheetHeader
          title="Sheet Title"
          buttons={<LinkButton text="Add something" icon="add" />}
        />
        <SheetContent>
          <p>Some content inside of the sheet...</p>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad aperiam
            assumenda commodi, consectetur, cupiditate et illo, laborum minus
            nihil officia porro provident quidem quisquam ratione recusandae sit
            ut? Perspiciatis, voluptas.
          </p>
          <p>
            A atque consequatur consequuntur cupiditate deleniti dolorem earum
            omnis, quasi quis ratione sed, tempora veritatis voluptas! A ad
            animi beatae dolor dolore doloremque dolores facilis ipsam, nesciunt
            odio quasi quisquam.
          </p>
        </SheetContent>
      </Sheet>
    );
    expect(rendered).toMatchSnapshot();
  });
});
