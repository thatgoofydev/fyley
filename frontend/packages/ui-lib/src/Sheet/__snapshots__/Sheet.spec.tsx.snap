// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sheet"
      >
        <div
          class="sheet__header"
        >
          <h1>
            Sheet Title
          </h1>
          <div>
            <div
              class="link-button"
              role="button"
            >
              <div
                class="icon"
                style="width: 14px; height: 14px;"
              >
                <svg
                  height="14"
                  viewBox="0 0 24 24"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 2a10 10 0 110 20 10 10 0 010-20zm0-2a12 12 0 100 24 12 12 0 000-24zm6 13h-5v5h-2v-5H6v-2h5V6h2v5h5v2z"
                  />
                </svg>
              </div>
              <span>
                Add something
              </span>
            </div>
          </div>
        </div>
        <div
          class="sheet__content"
        >
          <p>
            Some content inside of the sheet...
          </p>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad aperiam assumenda commodi, consectetur, cupiditate et illo, laborum minus nihil officia porro provident quidem quisquam ratione recusandae sit ut? Perspiciatis, voluptas.
          </p>
          <p>
            A atque consequatur consequuntur cupiditate deleniti dolorem earum omnis, quasi quis ratione sed, tempora veritatis voluptas! A ad animi beatae dolor dolore doloremque dolores facilis ipsam, nesciunt odio quasi quisquam.
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sheet"
    >
      <div
        class="sheet__header"
      >
        <h1>
          Sheet Title
        </h1>
        <div>
          <div
            class="link-button"
            role="button"
          >
            <div
              class="icon"
              style="width: 14px; height: 14px;"
            >
              <svg
                height="14"
                viewBox="0 0 24 24"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2a10 10 0 110 20 10 10 0 010-20zm0-2a12 12 0 100 24 12 12 0 000-24zm6 13h-5v5h-2v-5H6v-2h5V6h2v5h5v2z"
                />
              </svg>
            </div>
            <span>
              Add something
            </span>
          </div>
        </div>
      </div>
      <div
        class="sheet__content"
      >
        <p>
          Some content inside of the sheet...
        </p>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad aperiam assumenda commodi, consectetur, cupiditate et illo, laborum minus nihil officia porro provident quidem quisquam ratione recusandae sit ut? Perspiciatis, voluptas.
        </p>
        <p>
          A atque consequatur consequuntur cupiditate deleniti dolorem earum omnis, quasi quis ratione sed, tempora veritatis voluptas! A ad animi beatae dolor dolore doloremque dolores facilis ipsam, nesciunt odio quasi quisquam.
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
